zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/gateway/InclusiveGatewayTest.java

```diff
@@ -709,4 +714,541 @@ public void shouldEvaluateConditionWithoutDefaultFlow() {
+            .moveToNode("fork")
+            .serviceTask("task-2", s -> s.zeebeJobType("test"))
+            .exclusiveGateway("exclusive")
+            .conditionExpression("a > 1")
+            .connectTo("join")
+            .moveToNode("exclusive")
+            .conditionExpression("a <= 1")
+            .connectTo("task-2")
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+    final long processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).withVariable("a", 5).create();
+
+    // when
+    ENGINE.job().ofInstance(processInstanceKey).withType("test").complete();
+
+    // then
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETED)
+                .limitToProcessInstanceCompleted())
+        .extracting(e -> e.getValue().getElementId())
+        .containsOnlyOnce("join");
+  }
+
+  @Test
+  public void shouldJoinGatewayIfActiveElementWithLoopNoPathToGateway() {
+    // given
+    final var process =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent("start")
+            .parallelGateway("fork")
+            .task("task-1")
+            .inclusiveGateway("join")
+            .endEvent("end-1")
+            .moveToNode("fork")
+            .task("task-2")
+            .connectTo("join")
+            .moveToNode("fork")
+            .serviceTask("task-3")
+            .zeebeJobType("test")
+            .exclusiveGateway("exclusive")
+            .conditionExpression("a > 1")
+            .endEvent("end-2")
+            .moveToNode("exclusive")
+            .conditionExpression("a <= 1")
+            .connectTo("task-3")
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+    final long processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).withVariable("a", 5).create();
+
+    // when
+    ENGINE.job().ofInstance(processInstanceKey).withType("test").complete();
+
+    // then
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limitToProcessInstanceCompleted()
+                .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETED))
+        .extracting(e -> e.getValue().getElementId())
+        .containsOnlyOnce("join");
+  }
+
+  @Test
+  public void shouldJoinOnGatewayIfActiveLinkCatchEvent() {
+    // given
+    final ProcessBuilder processBuilder = Bpmn.createExecutableProcess(PROCESS_ID);
+    processBuilder
+        .startEvent()
+        .parallelGateway("fork")
+        .inclusiveGateway("join")
+        .endEvent()
+        .moveToNode("fork")
+        .intermediateThrowEvent("throw", b -> b.link("link"));
+
+    final BpmnModelInstance process =
+        processBuilder.linkCatchEvent("catch").link("link").connectTo("join").done();
```

Comment:
:wrench: We could use the fluent builder to add the link catch event by using `.moveToProcess()`.

Let's extend the process by a service task before the link throw event to verify that the process instance waits until the second token arrives at the inclusive gateway. 

```suggestion
    final var process =
        Bpmn.createExecutableProcess(PROCESS_ID)
            .startEvent()
            .parallelGateway("fork")
            .task("task-1")
            .inclusiveGateway("join")
            .endEvent()
            .moveToNode("fork")
            .serviceTask("task-2", s -> s.zeebeJobType("test"))
            .intermediateThrowEvent("throw", b -> b.link("link"))
            .moveToProcess(PROCESS_ID)
            .linkCatchEvent("catch")
            .link("link")
            .connectTo("join")
            .done();
```